mailbox/cassandra/src/main/java/org/apache/james/mailbox/cassandra/modules/CassandraQuotaModule.java

```diff
@@ -68,5 +70,26 @@ public interface CassandraQuotaModule {
             .withPartitionKey(CassandraGlobalMaxQuota.KEY, TEXT)
             .withColumn(CassandraGlobalMaxQuota.STORAGE, BIGINT)
             .withColumn(CassandraGlobalMaxQuota.MESSAGE, BIGINT))
+
+        .table(CassandraQuotaLimit.TABLE_NAME)
+        .comment("Holds quota limits.")
+        .options(options -> options
+            .withCaching(true, rows(CassandraConstants.DEFAULT_CACHED_ROW_PER_PARTITION)))
+        .statement(statement -> types -> statement
+            .withPartitionKey(CassandraQuotaLimit.QUOTA_COMPONENT, TEXT)
+            .withPartitionKey(CassandraQuotaLimit.QUOTA_SCOPE, TEXT)
+            .withPartitionKey(CassandraQuotaLimit.IDENTIFIER, TEXT)
+            .withClusteringColumn(CassandraQuotaLimit.QUOTA_TYPE, TEXT)
```

Comment:
Do you think we should open the door for a (future) potential API to get all quotas of a user (all of the Quota Components: jmap upload, mailbox... + Quota scopes...)?

The same case is an admin who would want to get all quotas of a domain in a single request.

With the current table design, I am afraid we would be closed to that usage. (always need QUOTA_COMPONENT + QUOTA_SCOPE + IDENTIFIER in `WHERE` condition).

I think Primary key = Partition Key (IDENTIFIER) + Clustering Key (QUOTA_COMPONENT + QUOTA_SCOPE + QUOTA_TYPE) would be more flexible.

WDYT?

Reply:
I thought we had agreement on this issue. If not, the whole design need to be reconsidered.

Reply:
I think with my proposed Cassandra table primary key, it does not only satisfy the current design but also opens for the above API.

Reply:
what if admin want to select quota by QUOTA_COMPONENT only?

Reply:
The clustering keys allow that level of grain fetch

Actually I read a bit fast at first but I agree with @quantranhong1999

Reply:
I have set QUOTA_SCOPE + IDENTIFIER as partition key, QUOTA_COMPONENT as clustering key

Reply:
Why do you need QUOTA_SCOPE in the partition key?

Reply:
IDENTIFIER alone does not make sense. For example, when identifier is 'abc', we need to know if it is domain name or user name

Reply:
Yes, you are right. We should know the scope while querying quota limits.

Reply:
Should we write DAO API + tests for the above use case also?